src/Extension/FluentExtension.php

```diff
@@ -654,6 +666,16 @@ public function onAfterDuplicate($original, $doWrite, $relations): void
                     SELECT ? AS \"RecordID\", \"Locale\", $fields_str
                     FROM \"$localisedTable\"
                     WHERE \"RecordID\" = ?", [$toID, $fromID]);
+
+            // Make sure to update any has_one IDs - otherwise the new localised table entry will
+            // have the has_one ID from the original record.
+            $fieldsToUpdate = [];
+            foreach (array_intersect($fields, $copyHasOneRelations) as $copyFieldName) {
+                $fieldsToUpdate[$copyFieldName] = $owner->$copyFieldName;
```

Comment:
Can you please add to this comment why the model (owner) in this has the correct data? I think that's the bit that's missing here.

Reply:
Done